src/javascript/app_2/App/Components/Elements/Calendar/calendar.jsx

```diff
@@ -0,0 +1,213 @@
+        this.setState({
+            calendar_date: default_date,
+            selected_date: '',
+        });
+    }
+
+    setToday = () => {
+        const { dateFormat, onSelect } = this.props;
+
+        const now = moment().utc().format(dateFormat);
+        this.setState({
+            calendar_date: now,
+            selected_date: now,
+            calendar_view: 'date',
+        });
+
+        if (onSelect) {
+            onSelect(now, true);
+        }
+    }
+
+    isPeriodDisabled = (date, unit) => {
+        const { maxDate, minDate } = this.props;
+
+        const start_of_period = moment(date).startOf(unit);
+        const end_of_period   = moment(date).endOf(unit);
+        return end_of_period.isBefore(moment(minDate))
+            || start_of_period.isAfter(moment(maxDate));
+    }
+
+    render() {
+        const { children, dateFormat, footer, id, maxDate, minDate, showTodayBtn } = this.props;
+        const { calendar_date, calendar_view, selected_date  } = this.state;
+
+        return (
+            <div id={id} className='calendar' value={selected_date}>
+                { children }
+                <CalendarHeader
+                    calendarDate={calendar_date}
+                    isPeriodDisabled={this.isPeriodDisabled}
+                    onClick={this.navigators}
+                    onSelect={this.calendarViews}
+                    view={calendar_view}
+                />
+                <CalendarPanel
+                    calendarDate={calendar_date}
+                    dateFormat={dateFormat}
+                    isPeriodDisabled={this.isPeriodDisabled}
+                    maxDate={maxDate}
+                    minDate={minDate}
+                    onClick={this.panelSelectors}
+                    selectedDate={selected_date}
+                    view={calendar_view}
+                />
+                <CalendarFooter
+                    footer={footer}
+                    onClick={this.setToday}
+                    showTodayBtn={showTodayBtn}
+                />
+            </div>
+        );
+    }
+}
+
+Calendar.defaultProps = {
+    dateFormat: 'YYYY-MM-DD',
+    minDate   : moment(0).utc().format('YYYY-MM-DD'),              // by default, minDate is set to Unix Epoch (January 1st 1970)
+    maxDate   : moment().utc().add(120, 'y').format('YYYY-MM-DD'), // by default, maxDate is set to 120 years after today
+};
+
+Calendar.propTypes = {
+    children       : PropTypes.object,
+    dateFormat     : PropTypes.string,
+    footer         : PropTypes.string,
+    onSelect       : PropTypes.func,
+    id             : PropTypes.string,
+    is_nativepicker: PropTypes.bool,
+    maxDate        : PropTypes.oneOfType([
+        PropTypes.object,
+        PropTypes.string,
+    ]),
+    minDate: PropTypes.oneOfType([
+        PropTypes.object,
+        PropTypes.string,
+    ]),
+    showTodayBtn: PropTypes.bool,
```

Comment:
`show_today_btn`?

Reply:
would be good if it reflects that it's a boolean by using `is_` or similar in the start as well